sbpy/activity/gas/core.py

```diff
@@ -1046,314 +1147,315 @@ def _make_radial_logspace_grid(self):
             badly so don't do it, dear reader
         """
         start_power = np.log10(
-            self.vmodel['collision_sphere_radius'].value * 2
+            self.vmr.collision_sphere_radius.value * 2
         )
-        end_power = np.log10(self.vmodel['max_grid_radius'].value)
+        end_power = np.log10(self.vmr.max_grid_radius.value)
         return np.logspace(
             start_power, end_power,
-            num=self.radial_points, endpoint=True
+            num=self.grid.radial_points, endpoint=True
         )
 
-    def _compute_fragment_density(self):
-        """ Computes the density of fragments as a function of radius
+    def _outflow_axis_sampling(self,
+                               x: np.float64,
+                               y: np.float64,
+                               theta: np.float64
+                               ) -> tuple[np.ndarray, np.ndarray]:
```

Comment:
I think for this to work, you'll need to use `Tuple` from the typing module, rather than the `tuple` object.

Reply:
This is what causes the tests to fail.